MekHQ/src/mekhq/gui/utilities/ExportXML.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * ExportXML.java
+ * 
+ * MegaMek - Copyright (C) 2018 - The MegaMek Team
+ * 
+ * This file is part of MekHQ.
+ * 
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq.gui.utilities;
+
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.util.ResourceBundle;
+
+import javax.swing.JTable;
+
+import megamek.common.logging.LogLevel;
+import mekhq.MekHQ;
+import mekhq.campaign.Campaign;
+import mekhq.campaign.CampaignOptions;
+import mekhq.campaign.parts.Part;
+import mekhq.campaign.personnel.Person;
+import mekhq.campaign.personnel.SkillType;
+import mekhq.campaign.personnel.SpecialAbility;
+import mekhq.gui.CampaignGUI;
+import mekhq.gui.GuiTabType;
+import mekhq.gui.PersonnelTab;
+import mekhq.gui.WarehouseTab;
+import mekhq.gui.model.PartsTableModel;
+
+
+public class ExportXML implements IExport
+{
+    @Override
+    public String exportTable(JTable table, File file) {
+        //TODO
+        return null;
+    }
+
+    @Override
+    public String exportFinances(File file, String format, Campaign c) {
+        //TODO
+        return null;
+    }
+
+    @Override
+    public String exportParts(File file, CampaignGUI gui) {
```

Comment:
IMHO most of this logic should really go into `PartsTableMouseAdapter` (this is not a generic method to export parts from a campagin: it exports the parts that are selected in the `partsTable` of the `WarehouseTab`) – the signature of this method could become something like `public String exportParts(File file, Part[] parts)` (or `Stream<Part> parts` / `Colleciton<Part> parts` / `Iterable<Part> parts, etc. anyone will do)

(same applies to other methods below)